File: matiasmasso/Mat.NET/DT2/DTOXec.vb
Public Class DTOXec
    Inherits DTOBaseGuid

    Property Emp As DTOEmp
    Public Property Id As Long
    Public Property Lliurador As DTOContact
    Public Property Iban As DTOIban
    Public Property XecNum As String
    Public Property Vto As Date = Date.MinValue
    Public Property Amt As DTOAmt
    Public Property Pnds As List(Of DTOPnd)
    Public Property Impagats As List(Of DTOImpagat)
    Public Property StatusCod As StatusCods
    Public Property FchRecepcio As Date
    Public Property CcaRebut As DTOCca
    Public Property CodPresentacio As ModalitatsPresentacio
    Public Property CcaPresentacio As DTOCca
    Public Property NBanc As DTOBanc
    Public Property CcaVto As DTOCca
    Public Property Format As Formats

    Public Enum StatusCods
        EnCartera
        EnCirculacio
        Vençut
    End Enum

    Public Enum Formats
        Xec
        Pagare
    End Enum

    Public Enum ModalitatsPresentacio
        NotSet
        A_la_Vista
        Al_Cobro
        Al_Descompte
    End Enum

    Public Sub New()
        MyBase.New()
    End Sub

    Public Sub New(oGuid As Guid)
        MyBase.New(oGuid)
    End Sub

    Shared Function Factory(oEmp As DTOEmp)
        Dim retval As New DTOXec
        With retval
            .Emp = oEmp
        End With
        Return retval
    End Function
End Class

File: matiasmasso/Mat.NET/DTO/DTOXec.cs
using System;
using System.Collections.Generic;

namespace DTO
{
    public class DTOXec : DTOBaseGuid
    {
        public DTOContact Lliurador { get; set; }
        public DTOIban Iban { get; set; }
        public string XecNum { get; set; }
        public DateTime Vto { get; set; } = DateTime.MinValue;
        public DTOAmt Amt { get; set; }
        public List<DTOPnd> Pnds { get; set; }
        public List<DTOImpagat> Impagats { get; set; }
        public StatusCods StatusCod { get; set; }
        public DateTime FchRecepcio { get; set; }
        public DTOCca CcaRebut { get; set; }
        public ModalitatsPresentacio CodPresentacio { get; set; }
        public DTOCca CcaPresentacio { get; set; }
        public DTOBanc NBanc { get; set; }
        public DTOCca CcaVto { get; set; }
        public Formats Format { get; set; }

        public enum StatusCods
        {
            EnCartera,
            EnCirculacio,
            Vençut
        }

        public enum Formats
        {
            Xec,
            Pagare
        }

        public enum ModalitatsPresentacio
        {
            NotSet,
            A_la_Vista,
            Al_Cobro,
            Al_Descompte
        }

        public DTOXec() : base()
        {
        }

        public DTOXec(Guid oGuid) : base(oGuid)
        {
        }

    }
}
